### ISSUE TITLE
Add --clipboard option to save recognized text directly to clipboard

### ISSUE THREAD

**Issue by shuanat** [tesseract-ocr/tesseract#4122]
### Your Feature Request

**Context**
Tesseract is a powerful OCR engine, but currently, it lacks the feature to save the recognized text to the clipboard directly. Users often have to save the text to a file and manually copy it into the clipboard, which could be cumbersome when working on tasks requiring frequent OCR operations.

**Feature Request**
I propose the addition of a --clipboard command-line option. When this option is used, Tesseract would directly save the recognized text to the clipboard instead of (or in addition to) saving it to a file.

**Example**
```
tesseract imagename outputbase --clipboard
or
tesseract imagename --clipboard
```

**Benefits**
Streamlining Operations: This will make the OCR process more efficient for tasks that require the result to be immediately used in other applications.
Ease of Use: Users won't have to manually copy text to the clipboard or write additional scripts to accomplish this task.
Cross-Platform: Ideally, this feature would be supported on all platforms where Tesseract can run.

**Additional Context**
This feature could be implemented using various clipboard libraries that support multiple platforms to ensure cross-compatibility.
An alternative to consider could be to allow the --clipboard option to work in conjunction with the output file option, allowing both to be specified if the user wishes to save the output in a file and clipboard simultaneously.

**Conclusion**
Adding the --clipboard option could make Tesseract more flexible and user-friendly, thereby increasing its adoption for a broader range of applications.

**Comment by stweil** [comment]
Tesseract already supports writing to STDOUT (`tesseract IMAGE -`). So it's possible to use software which writes STDIN to the clipboard. Here are some examples:

```
# Windows (untested)
tesseract IMAGE - | clip

# Linux (untested)
tesseract IMAGE - | xsel -i -b
tesseract IMAGE - | xclip -selection clipboard
tesseract IMAGE - | xclip -sel c

# macOS (tested)
tesseract IMAGE - | pbcopy
```

**Comment by stweil** [comment]
Implementing this feature is possible if we can use some portable code (it must work on Linux, macOS and Windows) with a compatible license. Is there such code available?

**Comment by stweil** [comment]
Do we really need to implement that functionality in Tesseract? Aren't the existing possibilities (see above) sufficient?

**Comment by shuanat** [comment]
👋 Hi @stweil,

Firstly, thanks for maintaining this invaluable project! The work you and the other contributors do makes Tesseract an indispensable tool in the OCR space.

After reading the feature request for a --clipboard option and your subsequent comments, I have to say I agree with your perspective for several reasons:

**1. UNIX Philosophy**
As a command-line tool, Tesseract aligns well with the UNIX philosophy of "Do one thing and do it well." The core function of Tesseract is OCR, and it does this exceptionally well. Additional functionalities like clipboard management could be handled by other specialized utilities (clip, xsel, xclip, pbcopy, etc.), keeping Tesseract focused and efficient.

**2. Existing Solutions**
As you rightly pointed out, the existing capability to pipe STDOUT to clipboard utilities already provides a flexible, powerful way to achieve the proposed feature's end goal. Implementing a --clipboard option might be redundant and could dilute the tool's focus.

```
# Windows
tesseract IMAGE - | clip

# Linux
tesseract IMAGE - | xsel -i -b

# macOS
tesseract IMAGE - | pbcopy
```

**3. Code Complexity**
Introducing a --clipboard option would necessitate adding new dependencies and libraries, which could result in compatibility issues and increased maintenance overhead. Keeping the codebase clean and focused lowers the barrier to entry for new contributors and makes the project easier to manage in the long term.

**4. Cross-Platform Challenges**
Clipboard management across different OS platforms is non-trivial. Even if a portable, license-compatible library is found, it adds another layer of complexity that could affect Tesseract's core functionality.

**5. Community Contributions**
For those who require a --clipboard option, it would be relatively simple to create a wrapper script around Tesseract that automatically pipes its output to the clipboard. This keeps the core codebase lean while allowing for such functionalities for those needing it.

In conclusion, Tesseract should focus on its core strength of providing robust, reliable OCR while leaving auxiliary tasks like clipboard management to other utilities or user-implemented scripts.

Thank you for considering my thoughts, and keep up the great work!
